src/storage/exec/IndexEdgeScanNode.cpp

```diff
@@ -0,0 +1,149 @@
+/* Copyright (c) 2021 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "storage/exec/IndexEdgeScanNode.h"
+namespace nebula {
+namespace storage {
+IndexEdgeScanNode::IndexEdgeScanNode(const IndexEdgeScanNode& node)
+    : IndexScanNode(node), edge_(node.edge_) {}
+IndexEdgeScanNode::IndexEdgeScanNode(RuntimeContext* context,
+                                     IndexID indexId,
+                                     const std::vector<cpp2::IndexColumnHint>& columnHint,
+                                     ::nebula::kvstore::KVStore* kvstore)
+    : IndexScanNode(context, "IndexEdgeScanNode", indexId, columnHint, kvstore) {
+  getIndex = std::function([this](std::shared_ptr<IndexItem>& index) {
+    auto env = this->context_->env();
+    auto indexMgr = env->indexMan_;
+    auto indexVal = indexMgr->getEdgeIndex(this->spaceId_, this->indexId_);
+    if (!indexVal.ok()) {
+      return ::nebula::cpp2::ErrorCode::E_INDEX_NOT_FOUND;
+    }
+    index = indexVal.value();
+    return ::nebula::cpp2::ErrorCode::SUCCEEDED;
+  });
+  getEdge = std::function([this](EdgeSchemas& edge) {
```

Comment:
Since the last change is so big, it is better to merge the Tag | Edge ScanNode into one. 
Tag or Edge can be identified from the indexItem, and the rest of the code should be the same.

Reply:
That's right, I did. In fact, all Scan logic is in IndexScanNode, IndexVertexScanNode and IndexEdgeScanNode are its two derived classes, which contain only a few schema-related processing logic